### PR TITLE
Add the CompleteCommand to complete an appointment

### PR DESCRIPTION
**Pull request by Kb-Tay**
Previously, the app cannot remove an appointment after completion. 

A new CompleteCommand is added to complete appointment either based on the FilteredList index or a specified date. All appointments with matching dates will then be removed. 

This gives user the flexibility to complete a specific appointment to a person or complete multiple appointments based on the date specified.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -191,6 +193,34 @@ public static Appointment parseAppointment(String appointmentName, String appoin
...
+        requireNonNull(date);
+        String trimmedDate = date.trim();
+
+        String dateValidation = "\\d{2}-\\d{2}-\\d{4}";
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+
+
+        if (!trimmedDate.matches(dateValidation)) {
+            throw new ParseException(CompleteCommand.MESSAGE_INVALID_DATE_FORMAT);
+        }
```
Comment (AlyssaPng) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:212]: What does this data validation checks if there is already a try and catch block?

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -191,6 +193,34 @@ public static Appointment parseAppointment(String appointmentName, String appoin
         return new Appointment(trimmedAppointmentName, appointmentDate);
     }
 
+    /**
+     * Parses a {@code String date} into a {@code LocalDate}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code date} is invalid.
+     */
+    public static LocalDate parseDate(String date) throws ParseException {
```
Comment (AlyssaPng) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:202]: Can check if the data is empty?

**Review by AlyssaPng — APPROVED** [review]
Looks good!! Can consider adding in more error handling, just to be more thorough.

File: src/main/java/seedu/address/logic/commands/CompleteCommand.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    public static final String COMMAND_WORD = "complete";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Completes an appointment identified by "
+            + "by the index number used in the displayed person list or "
+            + "by a specified date"
+            + "Parameters: INDEX(must be a positive integer) "
+            + "[" + PREFIX_APPOINTMENT_DATE + "Appointment Date] "
+            + "Example: " + COMMAND_WORD + " 1 ";
+    public static final String MESSAGE_COMPLETE_SUCCESS = "Appointments Completed!";
+    public static final String MESSAGE_NOT_COMPLETED = "At least one identifier must be provided.";
```
Comment (jylow) [inline src/main/java/seedu/address/logic/commands/CompleteCommand.java:35]: can you change this to provide more specific details on how to complete the command.

File: src/main/java/seedu/address/logic/commands/CompleteCommand.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+/**
+ * Completes and removes appointments from Person.
+ */
+public class CompleteCommand extends Command {
+
+    public static final String COMMAND_WORD = "complete";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Completes an appointment identified by "
+            + "by the index number used in the displayed person list or "
+            + "by a specified date"
```
Comment (jylow) [inline src/main/java/seedu/address/logic/commands/CompleteCommand.java:30]: should have a \n at the end of `specified date`

File: src/main/java/seedu/address/logic/commands/CompleteCommand.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        CompleteCommand otherEditCommand = (CompleteCommand) other;
+        return completeDescriptor.equals(otherEditCommand.completeDescriptor);
+    }
+
+    private static Person createPersonWithoutAppointment(Person personToEdit) {
+        return new Person(personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
+                personToEdit.getAddress(), personToEdit.getNextOfKinName(), personToEdit.getNextOfKinPhone(),
+                personToEdit.getFinancialPlans(), personToEdit.getTags(), NullAppointment.getNullAppointment());
+    }
+
```
Comment (jylow) [inline src/main/java/seedu/address/logic/commands/CompleteCommand.java:96]: Is there a reason this is here? seems like its not necessary unless testing?

**Review by jylow — COMMENTED** [review]
Im not sure what caused this but the system prints list of people like 5 times when the command is done.

Also i had assumed that the complete would have been done by completing all the appointments before the date and not only the date itself. I guess we can discuss this further next time to see which implementation makes more sense

File: src/main/java/seedu/address/logic/commands/CompleteCommand.java (hunk 5 above)
Comment (Kb-Tay) [reply src/main/java/seedu/address/logic/commands/CompleteCommand.java:96]: Person fields are all final, the only way to change Appointment to a NullAppointment is to create a copy of the Person and use the model setPerson method

**Pull request closed without merging**